browser/ui/views/infobars/brave_confirm_infobar.cc

```diff
@@ -25,14 +34,83 @@ std::unique_ptr<infobars::InfoBar> CreateBraveConfirmInfoBar(
 
 BraveConfirmInfoBar::BraveConfirmInfoBar(
     std::unique_ptr<BraveConfirmInfoBarDelegate> delegate)
-    : ConfirmInfoBar(std::move(delegate)) {
-  auto* delegate_ptr = GetBraveDelegate();
-  DCHECK(delegate_ptr);
+    : InfoBarView(std::move(delegate)) {
+  auto* delegate_ptr = GetDelegate();
+  label_ = AddChildView(CreateLabel(delegate_ptr->GetMessageText()));
+  label_->SetElideBehavior(delegate_ptr->GetMessageElideBehavior());
+
+  const auto create_button =
+      [this](ConfirmInfoBarDelegate::InfoBarButton type,
+             void (BraveConfirmInfoBar::*click_function)()) {
+        auto* button = AddChildView(std::make_unique<views::MdTextButton>(
+            base::BindRepeating(click_function, base::Unretained(this)),
+            GetDelegate()->GetButtonLabel(type)));
+        button->SetProperty(
+            views::kMarginsKey,
+            gfx::Insets::VH(ChromeLayoutProvider::Get()->GetDistanceMetric(
+                                DISTANCE_TOAST_CONTROL_VERTICAL),
+                            0));
+        return button;
+      };
+
+  const auto buttons = delegate_ptr->GetButtons();
+  if (buttons & ConfirmInfoBarDelegate::BUTTON_OK) {
+    ok_button_ = create_button(ConfirmInfoBarDelegate::BUTTON_OK,
+                               &BraveConfirmInfoBar::OkButtonPressed);
+    ok_button_->SetProminent(true);
+    ok_button_->SetImageModel(
+        views::Button::STATE_NORMAL,
+        delegate_ptr->GetButtonImage(ConfirmInfoBarDelegate::BUTTON_OK));
+    ok_button_->SetEnabled(
+        delegate_ptr->GetButtonEnabled(ConfirmInfoBarDelegate::BUTTON_OK));
+    ok_button_->SetTooltipText(
+        delegate_ptr->GetButtonTooltip(ConfirmInfoBarDelegate::BUTTON_OK));
+  }
+
+  if (buttons & ConfirmInfoBarDelegate::BUTTON_CANCEL) {
+    cancel_button_ = create_button(ConfirmInfoBarDelegate::BUTTON_CANCEL,
+                                   &BraveConfirmInfoBar::CancelButtonPressed);
+    if (features::IsChromeRefresh2023()) {
+      cancel_button_->SetStyle(ui::ButtonStyle::kTonal);
+    }
+    if (buttons == ConfirmInfoBarDelegate::BUTTON_CANCEL ||
+        delegate_ptr->IsProminent(ConfirmInfoBarDelegate::BUTTON_CANCEL)) {
+      cancel_button_->SetProminent(true);
+    }
+    cancel_button_->SetImageModel(
+        views::Button::STATE_NORMAL,
+        delegate_ptr->GetButtonImage(ConfirmInfoBarDelegate::BUTTON_CANCEL));
+    cancel_button_->SetEnabled(
+        delegate_ptr->GetButtonEnabled(ConfirmInfoBarDelegate::BUTTON_CANCEL));
+    cancel_button_->SetTooltipText(
+        delegate_ptr->GetButtonTooltip(ConfirmInfoBarDelegate::BUTTON_CANCEL));
+  }
+
+  if (buttons & ConfirmInfoBarDelegate::BUTTON_EXTRA) {
+    extra_button_ = create_button(ConfirmInfoBarDelegate::BUTTON_EXTRA,
+                                  &BraveConfirmInfoBar::ExtraButtonPressed);
+    if (features::IsChromeRefresh2023()) {
+      extra_button_->SetStyle(ui::ButtonStyle::kTonal);
+    }
```

Comment:
Forgot to mention. I think we don't need this as we'll not have different UI for this feature.